### ISSUE TITLE
@PolyAll should apply only to relevant types

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#1654]
Some annotations, like `@Initialized` and `@Interned`, only apply to certain types (non-primitive ones).
When `@PolyAll` is written on a primitive type, it should mean "polymorphic for all types that apply to this type".

Here is an example of code that fails under the Nullness Checker (it is checked in as `checker/tests/nullness/Issue1654.java`:

```java
  public static /*@PolyAll*/ int[] fn_compose(int[] a, /*@PolyAll*/ int[] b) {
    /*@PolyAll*/ int[] result = new int[a.length];
    for (int i = 0; i < a.length; i++) {
      result[i] = b[a[i]];
    }
    return result;
  }
```

It should pass.

Also, the error message should be improved.  Right now the error message is:

```
warning: [type.invalid] [@Initialized, @PolyAll] may not be applied to the type "@Initialized @PolyAll int"
```

but this is confusing.  What's really illegal is applying `@Initialized` to `int`.
(The message "`@Initialized` may not be applied to `@Initialized int`" at first led me to believe that `@Initialized` was incorrectly being applied twice.  Other users may not come to the same conclusion, but the message should still be improved.)
